src/hotspot/share/code/nmethod.hpp

```diff
@@ -221,57 +238,37 @@ class nmethod : public CodeBlob {
   // Offset of the unwind handler if it exists
   int _unwind_handler_offset;
 
-  int _consts_offset;
-  int _stub_offset;
-  int _oops_offset;                       // offset to where embedded oop table begins (inside data)
-  int _metadata_offset;                   // embedded meta data table
-  int _scopes_data_offset;
-  int _scopes_pcs_offset;
-  int _dependencies_offset;
-  int _handler_table_offset;
-  int _nul_chk_table_offset;
+  uint16_t _skipped_instructions_size;
+
+  // _oops_offset == _data_offset,  offset where embedded oop table begins (inside data)
+  uint16_t _metadata_offset; // embedded meta data table
+  uint16_t _dependencies_offset;
+  uint16_t _scopes_pcs_offset;
+  int      _scopes_data_offset;
+  int      _handler_table_offset;
+  int      _nul_chk_table_offset;
 #if INCLUDE_JVMCI
-  int _speculations_offset;
-  int _jvmci_data_offset;
+  int      _speculations_offset;
+  int      _jvmci_data_offset;
 #endif
-  int _nmethod_end_offset;
-  int _skipped_instructions_size;
 
   // location in frame (offset for sp) that deopt can store the original
   // pc during a deopt.
   int _orig_pc_offset;
 
-  int _compile_id;                        // which compilation made this nmethod
-
-  CompilerType _compiler_type;            // which compiler made this nmethod (u1)
-
-  bool _is_unlinked;
+  int          _compile_id;            // which compilation made this nmethod
```

Comment:
NIT: are these fields always needed?

Reply:
Yes, they are needed for debugging issues. They are important for error reporting, logs and events recording.  And they do not take much space: CompLevel and CompilerType are one byte size.